kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/OperationsResource.scala

```diff
@@ -81,4 +83,42 @@ private[v1] class OperationsResource extends ApiRequestContext {
           s"for operation handle $operationHandleStr")
     }
   }
+
+  @ApiResponse(
+    responseCode = "200",
+    content = Array(new Content(
+      mediaType = MediaType.APPLICATION_JSON)),
+    description =
+      "get result set metadata")
+  @GET
+  @Path("{operationHandle}/resultsetmetadata")
+  def getResultSetMetadata(
+      @PathParam("operationHandle") operationHandleStr: String): ResultSetMetaData = {
+    try {
+      val operationHandle = parseOperationHandle(operationHandleStr)
+      ResultSetMetaData(
+        backendService.getResultSetMetadata(operationHandle).getColumns.asScala.map(c => {
+          val tPrimitiveTypeEntry = c.getTypeDesc.getTypes.get(0).getPrimitiveEntry
+          var precision = 0
+          var scale = 0
+          if (tPrimitiveTypeEntry.getTypeQualifiers != null) {
+            val qualifiers = tPrimitiveTypeEntry.getTypeQualifiers.getQualifiers
+            val defaultValue = TTypeQualifierValue.i32Value(0);
+            precision = qualifiers.getOrDefault("precision", defaultValue).getI32Value
+            scale = qualifiers.getOrDefault("scale", defaultValue).getI32Value
+          }
+          ColumnDesc(
+            c.getColumnName,
+            tPrimitiveTypeEntry.getType.toString,
+            c.getPosition,
+            precision,
+            scale,
+            c.getComment)
+        }))
+    } catch {
+      case NonFatal(_) =>
+        throw new NotFoundException(
+          s"Error getting result set metadata for operation handle $operationHandleStr")
```

Comment:
can we also add a unit test for negative case ?

Reply:
The negative case about `operationHandle` have been tested in other api ut before, the scene is the same.
And if metadata is empty,it won't throw exception.